mesonbuild/interpreter.py

```diff
@@ -3394,6 +3394,12 @@ def unpack_env_kwarg(self, kwargs) -> build.EnvironmentVariables:
         envlist = kwargs.get('env', EnvironmentVariablesHolder())
         if isinstance(envlist, EnvironmentVariablesHolder):
             env = envlist.held_object
+        elif isinstance(envlist, dict):
+            FeatureNew('environment dictionary', '0.52.0').use(self.subproject)
```

Comment:
Oh, I notice that I forgot `FeatureNew()` with configuration_data(), was added in 0.49.0. Mind to add that while at it please? :D

Reply:
ok